src/index.ts

```diff
@@ -76,61 +44,79 @@ const addKeyValuePairToObject = (obj: any, keyValuePair: string): any => {
   const value = values.join("="); // in case there were values with = in them
   /* eslint-disable no-param-reassign */
   // this is necessary because we modify the accumulator in .reduce
-  obj[key.trim()] = value.trim();
+  obj[key.trim()] = trimValue(value);
 
   return obj;
 };
 
+const formatDnsGenericRecord = ({ a, v, p }: { [key: string]: string }) => {
+  return {
+    algorithm: a,
+    publicKey: p,
+    version: v,
+  };
+};
+
+export const queryDns = async (domain: string): Promise<IDNSQueryResponse> => {
+  const { data } = await axios.get<IDNSQueryResponse>(`https://dns.google/resolve?name=${domain}&type=TXT`);
+  return data;
+};
+
 /**
  * Parses one openattestation DNS-TXT record and turns it into an OpenAttestationsDNSTextRecord object
  * @param record e.g: '"openatts net=ethereum netId=3 addr=0x0c9d5E6C766030cc6f0f49951D275Ad0701F81EC"'
  */
-export const parseOpenAttestationRecord = (record: string): OpenAttestationDNSTextRecord => {
+export const parseOpenAttestationRecord = (record: string): GenericObject => {
   trace(`Parsing record: ${record}`);
   const keyValuePairs = record.trim().split(" "); // tokenize into key=value elements
-  const recordObject = {} as OpenAttestationDNSTextRecord;
+  const recordObject = {} as GenericObject;
   // @ts-ignore: we already checked for this token
   recordObject.type = keyValuePairs.shift();
-  keyValuePairs.reduce<OpenAttestationDNSTextRecord>(addKeyValuePairToObject, recordObject);
+  keyValuePairs.reduce<GenericObject>(addKeyValuePairToObject, recordObject);
   return recordObject;
 };
 
 /**
  * Currying function that applies a given dnssec result
  */
-const applyDnssecResults = (dnssecStatus: boolean) => (
-  record: OpenAttestationDNSTextRecord
-): OpenAttestationDNSTextRecord => {
+const applyDnssecResults = <T>(dnssecStatus: boolean) => (record: T): T => {
   return { ...record, dnssec: dnssecStatus };
 };
 
-/**
- * Returns true if the given object passes runtype validation for OpenAttestationDNSTextRecord
- * Turn on debug log to see exact validation failure messages if necessary
- * @param record An object that may conform to the OpenAttestationDNSTextRecord shape
- */
-export const isWellFormedOpenAttestationRecord = (record: OpenAttestationDNSTextRecord) => {
-  try {
-    OpenAttestationDNSTextRecord.check(record);
-    return true;
-  } catch (e) {
-    error(e);
-    return false;
-  }
+const parseOpenAttestationRecords = (recordSet: IDNSRecord[] = []): GenericObject[] => {
```

Comment:
would be nice to have a comment block here showing an example of a input as it's not obvious from the code